### DISCUSSION TITLE
pb_hooks onAfterMigrations?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#3692] [Q&A]
I need my hook to run after migrations have finished. `onAfterBootstrap` is too early, is there another option?

**Comment by ganigeorgiev** [comment]
There is no such JS `pb_hook` at the moment. One slightly hacky workaround could be to attach to the original Go `OnBeforeServe` with `$app.onBeforeServe().add((e) => { ... })` (see also https://github.com/pocketbase/pocketbase/discussions/3220).

**Reply by benallfree** [reply]
For now I solved it this way, which probably isn't perfect... my thinking is that when the first request comes in, I can take some quick action and the skip it for future requests. ideally an onAfterMigrate or even onServe would be great.

```js
routerUse((next) => {
  return (c) => {
    const globals = require(`${__hooks}/pockethost-globals.js`)
    if (globals.hasRun) {
      console.log(`***already ran`)
      return next(c)
    }
    console.log(`***running for the first time`)
    globals.hasRun = true
    // do stuff
})
```
